src/SUAPI-CppWrapper/model/Curve.cpp

```diff
@@ -47,7 +49,10 @@ Curve::~Curve(){
 
 SUCurveRef Curve::create_curve(std::vector<Edge>& edges, SUResult &result) {
   SUCurveRef curve_ref = SU_INVALID;
-  result = SUCurveCreateWithEdges(&curve_ref, edges[0], edges.size());
+  std::vector<SUEdgeRef> refs{};
```

Comment:
Did you try this? `std::transform` expects the target container to be as large as the source container. This should be `std::vector<SUEdgeRef> refs(edge.size(), SU_INVALID);`

(Worth a test unit for this.)

Reply:
No, this whole class has been neglected.  The commit was a little bit Find and Replace.  Will fix

Reply:
This is where it would be good to add test units.